packages/node_modules/@node-red/runtime/lib/flows/Flow.js

```diff
@@ -404,6 +415,57 @@ class Flow {
         return this.activeNodes;
     }
 
+    /**
+     * Get a flow setting value deined in this flow. 
+     * @param  {String} key
+     * @return {Object} result containinig val property or null
+     */
+    getFlowSetting(name) {
+        const flow = this.flow; 
+        if (flow.env) {
+            if (!name.startsWith("$parent.")) {
```

Comment:
I do not remember us discussing `$parent` with env vars. I can't see it mentioned in the designs or slides - but I could be missing something.

What is the use-case for needing it?

It is needed with context inside subflows to allow a subflow to read/write from the parent flow context.

But as env vars are inherited (and read-only) then there is not the same requirement - you can read the parent env vars by just using it.

The only scenario it enables is it allows a node to read the parent env var that group may have changed the value of. I cannot immediately see any scenario where that would be useful, given the static nature of env vars.

We do not want to add any more complexity than necessary.

Reply:
I would like to hear your opinion on this before I make any modifications.
In the current dev branch code, `$parent` can be used to move the environment variable search position up to enclosing subflow.  This fix is intended to make it match the new search order.

Is it your opinion that `$parent` starts the environment variable lookup from the outer subflow?
When using `$parent` in the outermost SUBFLOW, the current implementation returns the value of the OS environment variable, but should this also be left without referencing tab?

Reply:
Sorry - I forgot we already had `$parent` support for env vars. Given that, you are right to ensure it behaves as a user would expect once groups/flows have env vars.